### PR TITLE
T167886 sofort bank transfer code

### PR DESCRIPTION
**Pull request by wiese**
* assigning bank transfer code on donation for sofort, too
* setting sofort donation initial status to PROMISE

~~Composer needs to be pointed to a stable version of Store before merge.~~

### CODE REVIEW

**Comment by wiese** [comment]
Using store's master HEAD, still needs switch to pin.

File: contexts/DonationContext/tests/Unit/DataAccess/DoctrineDonationRepositoryTest.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+		$method = new ReflectionMethod( $repo, 'getBankTransferCode' );
+		$method->setAccessible( true );
+
+		$paymentMethod = new SofortPayment( 'fff' );
+		$this->assertSame( 'fff', $method->invoke( $repo, $paymentMethod ) );
+		$paymentMethod = new BankTransferPayment( 'ggg' );
+		$this->assertSame( 'ggg', $method->invoke( $repo, $paymentMethod ) );
+		$paymentMethod = new CreditCardPayment( null );
+		$this->assertSame( '', $method->invoke( $repo, $paymentMethod ) );
```
Comment (JeroenDeDauw) [inline contexts/DonationContext/tests/Unit/DataAccess/DoctrineDonationRepositoryTest.php]: I would much rather have a test that checks if a SofortPayment gets stored correctly, which will catch the not setting of the bank transfer code as well.

**Comment by wiese** [comment]
Updated.

**Comment by JeroenDeDauw** [comment]
I think this is linking the wrong issue, since T167886 is just about the store. Not finding a fitting one though

**Comment by JeroenDeDauw** [comment]
ugggh need to also update th lock file... can't easily do that on this box

**Comment by wiese** [comment]
Use stable version of the store (composer.json & composer.lock)

**Pull request merged**